### PR TITLE
added update and delete APIs

### PR DESCRIPTION
**Pull request by camperjett**
added APIs for Admin APIs
{

		adminAPI.Post("/mess", handler.CreateMess)
		adminAPI.Get("/mess", handler.GetMess)
		adminAPI.Put("/mess", handler.UpdateMess)
		adminAPI.Delete("/mess", handler.DeleteMess)

		adminAPI.Post("/hostel", handler.CreateHostel)
		adminAPI.Get("/hostel", handler.GetHostel)
		adminAPI.Put("/hostel", handler.UpdateHostel)
		adminAPI.Delete("/hostel", handler.DeleteHostel)

		adminAPI.Post("/meal", handler.CreateMeal)
		adminAPI.Get("/meal", handler.GetMeal)
		adminAPI.Put("/meal", handler.UpdateMeal)
		adminAPI.Delete("/meal", handler.DeleteMeal)

		adminAPI.Post("/meal-type", handler.CreateMealType)
		adminAPI.Get("/meal-type", handler.GetMealType)
		adminAPI.Put("/meal-type", handler.UpdateMealType)
		adminAPI.Delete("/meal-type", handler.DeleteMealType)

		adminAPI.Post("/menu-item", handler.CreateMenuItem)
		adminAPI.Get("/menu-item", handler.GetMenuItem)
		adminAPI.Put("/menu-item", handler.UpdateMenuItem)
		adminAPI.Delete("/menu-item", handler.DeleteMenuItem)

		adminAPI.Post("/room", handler.CreateRoom)

		adminAPI.Post("/user", handler.CreateUser)
		adminAPI.Get("/user", handler.GetUser)
		adminAPI.Put("/user", handler.UpdateUser)
		adminAPI.Delete("/user", handler.DeleteUser)

		adminAPI.Get("/user/populated", handler.GetUserPopulated)
		adminAPI.Post("/user/manage/hostel-mess", handler.ManageHostelMess)
	}

### CODE REVIEW

File: handler/mess.go [hunk 1]
```diff
@@ -43,6 +43,49 @@ func (handler *Handler) GetMess(ctx *fiber.Ctx) error {
...
+
+func (handler *Handler) DeleteMess(ctx *fiber.Ctx) error {
+	messID, errObjID := primitive.ObjectIDFromHex(ctx.Get("messID"))
+	if errObjID != nil {
+		return ctx.Status(http.StatusBadRequest).JSON(fiber.Map{"error": "Invalid mess ID"})
+	}
+
+	collection := handler.MongikClient.MongoClient.Database(constants.DB).Collection(constants.COLLECTION_MESSES)
+	if result, err := collection.DeleteOne(ctx.Context(), bson.M{"_id": messID}); err != nil {
+		return ctx.Status(http.StatusInternalServerError).JSON(fiber.Map{"error": "Failed to delete mess"})
```
Comment (shalearkane) [inline handler/mess.go]: send err.Error() in the error part

File: handler/mess.go [hunk 2]
```diff
@@ -43,6 +43,49 @@ func (handler *Handler) GetMess(ctx *fiber.Ctx) error {
 	return ctx.JSON(interfaces.GetGenericResponse(true, "Found Mess with the given Mess ID", Mess, nil))
 }
 
+func (handler *Handler) UpdateMess(ctx *fiber.Ctx) error {
+	var updateData models.Mess
+	messID, errObjID := primitive.ObjectIDFromHex(ctx.Get("messID"))
+	if errObjID != nil {
+		return ctx.Status(http.StatusBadRequest).JSON(fiber.Map{"error": "Invalid mess ID"})
```
Comment (shalearkane) [inline handler/mess.go]: send errObjID back along with this message

File: handler/mess.go [hunk 3]
```diff
@@ -43,6 +43,49 @@ func (handler *Handler) GetMess(ctx *fiber.Ctx) error {
...
+		return ctx.Status(http.StatusBadRequest).JSON(fiber.Map{"error": err.Error()})
+	}
+
+	collection := handler.MongikClient.MongoClient.Database(constants.DB).Collection(constants.COLLECTION_MESSES)
+	update := bson.M{
+		"$set": updateData,
+	}
+	result, err := collection.UpdateOne(ctx.Context(), bson.M{"_id": messID}, update)
+	if err != nil {
+		return ctx.Status(http.StatusInternalServerError).JSON(fiber.Map{"error": "Failed to update mess"})
```
Comment (shalearkane) [inline handler/mess.go]: send err.Error() along with this message

File: handler/mess.go [hunk 4]
```diff
@@ -43,6 +43,49 @@ func (handler *Handler) GetMess(ctx *fiber.Ctx) error {
...
+		return ctx.Status(http.StatusNotFound).JSON(fiber.Map{"error": "Mess not found"})
+	}
+
+	return ctx.JSON(interfaces.GetGenericResponse(true, "Mess updated successfully", nil, nil))
+}
+
+func (handler *Handler) DeleteMess(ctx *fiber.Ctx) error {
+	messID, errObjID := primitive.ObjectIDFromHex(ctx.Get("messID"))
+	if errObjID != nil {
+		return ctx.Status(http.StatusBadRequest).JSON(fiber.Map{"error": "Invalid mess ID"})
```
Comment (shalearkane) [inline handler/mess.go]: send errObjID.Error() along with this message

**Comment by camperjett** [comment]
@shalearkane PTAL.

**Review by shalearkane — APPROVED** [review]

**Pull request merged**